File: codeflow-core/test-fixtures/sample-ts/service.ts
import { BaseService } from "./base-service.js";

/**
 * Process data and return a formatted result.
 */
export function processData(input: string): string {
  return `processed: ${input}`;
}

export function calculateSum(a: number, b: number): number {
  return a + b;
}

/**
 * UserService handles user operations.
 */
export class UserService extends BaseService {
  private dbHost: string;

  constructor(host: string) {
    super();
    this.dbHost = host;
  }

  getUser(id: number): string {
    const data = processData("user");
    return data;
  }

  saveUser(name: string): boolean {
    const formatted = processData(name);
    return formatted.length > 0;
  }
}

export class TaskService extends BaseService {
  createTask(title: string): string {
    const result = processData(title);
    return result;
  }

  run(): boolean {
    this.connect();
    return true;
  }
}

const svc = new UserService("localhost");
const user = svc.getUser(1);

File: codeflow-core/test-fixtures/sample-python/service.py
"""
User service module that handles user operations.
"""

import os
import json


def load_config(path: str) -> dict:
    """Load configuration from a JSON file."""
    with open(path) as f:
        return json.load(f)


def process_data(input_str: str) -> str:
    """Process input string and return formatted result."""
    return f"processed: {input_str}"


class UserService:
    """Service for managing users."""

    def __init__(self, db_host: str):
        self.db_host = db_host

    def get_user(self, user_id: int) -> str:
        """Get a user by ID."""
        data = process_data("user")
        return data

    def save_user(self, name: str) -> bool:
        """Save a user to the database."""
        formatted = process_data(name)
        return len(formatted) > 0


class BaseService:
    """Base service with common functionality."""

    def connect(self) -> bool:
        """Connect to the service."""
        return True


class TaskService(BaseService):
    """Service for managing tasks."""

    def create_task(self, title: str) -> str:
        """Create a new task."""
        result = process_data(title)
        return result

    def run(self) -> bool:
        """Run the task service."""
        self.connect()
        return True
